### ISSUE TITLE
binary cross entropy  - inappropriate for xy_loss

### ISSUE THREAD

**Issue by pabloe4993** [qqwweee/keras-yolo3#459]
The use of the BCE loss function on xy is inappropriate .
The BCE is relevant when we use the sigmoid function as the probability of belonging to some class, as opposed to usage in yolo v3 (xy_loss), where the use of the sigmoid function is to predict the center coordinates of the box relative to the location of filter.
for example if the true xy center is [0.5,0.5] and the prediction is perfect ([0.5,0.5]) the BCE loss is 0.6931472 (for each coordinate) while is should be zero.
The appropriate loss is square distance.

https://github.com/qqwweee/keras-yolo3/blob/e6598d13c703029b2686bc2eb8d5c09badf42992/yolo3/model.py#L398-L399

**Comment by 791136190** [comment]
I also have the same question, the loss does not need to be 0, only need to have a minimum. But from the image point of view, when the minimum value is obtained, the label and the predicted value are not equal.
![image](https://user-images.githubusercontent.com/27891060/58004326-94a6cd80-7b15-11e9-9b43-b1148558b7ef.png)

**Comment by Starboy-at-earth** [comment]
This usage confuses me a lot. But why the code still work may prove the author is right. I should think the usage again.
